src/planscape/forsys/forsys_request_params.py

```diff
@@ -350,3 +304,152 @@ def _get_default_planning_area(self) -> MultiPolygon:
+        self.planning_area = self._get_planning_area(self.huc12_names)
+
+    def get_priority_weights_dict(self) -> dict[str, float]:
+        return ForsysGenerationRequestParams.get_priority_weights_dict(self)
+
+    def _get_planning_area(self, huc12_names: list[str]) -> GEOSGeometry:
+        huc12s = BoundaryDetails.objects.filter(
+            boundary_id=self._HUC12_ID).filter(shape_name__in=huc12_names)
+        polygons = [huc12.geometry for huc12 in huc12s]
+        return merge_polygons(polygons, 0)
+
+
+class SilverCreekForsysGenerationParams(ForsysGenerationRequestParamsFromHuc12):
+    _DEFAULT_HUC12_NAMES = [
+        'Little Silver Creek-Silver Creek',
+        'South Fork Rubicon River',
+        'Jones Fork Silver Creek',
+        'Brush Creek-South Fork American River',
+        'South Fork Silver Creek'
+    ]
+
+    def get_priority_weights_dict(self) -> dict[str, float]:
+        return ForsysGenerationRequestParams.get_priority_weights_dict(self)
+
+
+class CowCreekForsysGenerationParams(ForsysGenerationRequestParamsFromHuc12):
+    _DEFAULT_HUC12_NAMES = [
+        'Upper Old Cow Creek',
+        'Lower Old Cow Creek',
+        'Upper South Cow Creek',
+        'Lower South Cow Creek',
+        'Glendenning Creek',
+        'Clover Creek',
+        'Oak Run Creek'
+    ]
+
+    def get_priority_weights_dict(self) -> dict[str, float]:
+        return ForsysGenerationRequestParams.get_priority_weights_dict(self)
+
+
+class MiddleForkForsysGenerationParams(ForsysGenerationRequestParamsFromHuc12):
+    _DEFAULT_HUC12_NAMES = [
+        'Middle Fork Bishop Creek',
+        'South Fork Bishop Creek',
+        'Evolution Creek',
+        'Headwaters Middle Fork Kings River',
+        'Goddard Creek',
+        'Goddard Canyon-South Fork San Joaquin River',
+        'Upper Middle Fork Kings River'
+    ]
+
+    def get_priority_weights_dict(self) -> dict[str, float]:
+        return ForsysGenerationRequestParams.get_priority_weights_dict(self)
+
+
+# Returns ForsysRankingRequestParams based on url parameter value for the
+# parameter name in _URL_REQUEST_PARAMS_TYPE.
+def get_ranking_request_params(
+        params: QueryDict) -> ForsysRankingRequestParams:
+    type = ForsysRankingRequestParamsType(
+        int(params.get(_URL_REQUEST_PARAMS_TYPE, 0)))
+    if type == ForsysRankingRequestParamsType.DATABASE:
+        return ForsysRankingRequestParamsFromDb(params)
+    elif type == ForsysRankingRequestParamsType.URL_WITH_DEFAULTS:
+        return ForsysRankingRequestParamsFromUrlWithDefaults(params)
+    else:
+        raise Exception("ranking request type was not recognized")
+
+
+# Returns ForsysGenerationRequestParams based on url parameter value for the
+# parameter name in _URL_REQUEST_PARAMS_TYPE.
+def get_generation_request_params(
+        request: HttpRequest) -> ForsysGenerationRequestParams:
+    params = request.GET
+    type = ForsysGenerationRequestParamsType(int(
```

Comment:
these types a little bit hard to interpret without looking at the classes. i would suggest something like this for consistency:

DATABASE
ALL_DEFAULTS (we know that use of defaults implies URL) 
HUC12s_WITH_DEFAULTS
SILVER_CREEK_HUC12s_WITH_DEFAULTS
etc.

Reply:
Done.